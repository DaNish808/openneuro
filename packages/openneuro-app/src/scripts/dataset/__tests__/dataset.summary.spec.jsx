import React from 'react'
import { shallow } from 'enzyme'
import Summary from '../dataset.summary.jsx'

const defProps = {
  summary: {
    sessions: [],
    totalFiles: 1,
    size: 5824,
    subjects: ['01', '02'],
    tasks: ['psuedotask'],
    modalities: ['MRI'],
  },
  minimal: false,
}

describe('dataset/Summary', () => {
  it('renders successfully', () => {
    const wrapper = shallow(<Summary {...defProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('applies the minimal-summary class with the minimal prop', () => {
    const minimalProps = Object.assign({}, defProps, { minimal: true })
    const wrapper = shallow(<Summary {...minimalProps} />)
    expect(wrapper.find('.minimal-summary')).toHaveLength(1)
  })
})
