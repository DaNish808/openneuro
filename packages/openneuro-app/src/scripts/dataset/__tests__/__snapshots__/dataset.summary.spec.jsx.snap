// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/Summary renders successfully 1`] = `
<div
  className="clearfix"
>
  <div>
    <hr />
    <h5>
      <b>
        <span>
          <strong>
             
            File
            : 
          </strong>
          1
        </span>
        , 
        <span>
          <strong>
            Size: 
          </strong>
          5.69KB
        </span>
        , 
        <span>
          <strong>
             
            Subjects
            : 
          </strong>
          2
        </span>
        , 
        <span>
          <strong>
            Session
            : 
          </strong>
          1
        </span>
      </b>
    </h5>
    <h5>
      <span>
        <b>
          Available
        </b>
         
        <b>
          Tasks
        </b>
         : 
        psuedotask
      </span>
    </h5>
    <h5>
      <span>
        <b>
          Available
        </b>
         
        <b>
          Modalities
        </b>
         : 
        MRI
      </span>
    </h5>
  </div>
</div>
`;
